### PR TITLE
23279: Implementation of integration tests for ProductAlert Stock notifications 

### PR DESCRIPTION
**Pull request by yuriichayka**
### Description (*)

Added two test classes along with one fixture:
- Magento\ProductAlert\Controller\Add\StockTest.php
- Magento\ProductAlert\Controller\Unsubscribe\StockTest.php
- dev/tests/integration/testsuite/Magento/ProductAlert/_files/customer_unsubscribe_stock.php
### Fixed Issues (if relevant)

1. magento/magento2#23279: Integration test for unsubscribing from 'Back in stock notification'

### Manual testing scenarios (*)

 Not required

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: dev/tests/integration/testsuite/Magento/ProductAlert/_files/customer_unsubscribe_stock.php [hunk 1]
```diff
@@ -0,0 +1,33 @@
+<?php
```
Comment (phoenix128) [inline dev/tests/integration/testsuite/Magento/ProductAlert/_files/customer_unsubscribe_stock.php:1]: You should also write `dev/tests/integration/testsuite/Magento/ProductAlert/_files/customer_unsubscribe_stock_rollback.php` file cleaning the inserted entries.

**Review by phoenix128 — CHANGES_REQUESTED** [review]
Hello @yuriichayka ,
thank you for your contribution, would you please check my comments?

File: dev/tests/integration/testsuite/Magento/ProductAlert/_files/customer_unsubscribe_stock.php (hunk 1 above)
Comment (yuriichayka) [reply dev/tests/integration/testsuite/Magento/ProductAlert/_files/customer_unsubscribe_stock.php:1]: Hello, thanks for the suggestion.
The file was added.

**Comment by sidolov** [comment]
@magento run all tests

File: dev/tests/integration/testsuite/Magento/ProductAlert/_files/customer_unsubscribe_stock_rollback.php [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+$objectManager = Bootstrap::getObjectManager();
+$resource = $objectManager->get(Stock::class);
+
+$productRepository = $objectManager->create(ProductRepositoryInterface::class);
+$productId = $productRepository->get('simple-out-of-stock')->getId();
+
+$resource->getConnection()->delete(
+    $resource->getMainTable(),
+    'product_id = ' . $productId
```
Comment (phoenix128) [inline dev/tests/integration/testsuite/Magento/ProductAlert/_files/customer_unsubscribe_stock_rollback.php]:
```
$resource->getConnection()->delete(
    $resource->getMainTable(),
    ['product_id = ?' => $productId]
)
```

Your code is absolutely safe, no security threats here, but using string concatenation in queries is bad behavior from the security perspective.

**Review by phoenix128 — CHANGES_REQUESTED** [review]
Thank you for the required changes, please check my last comment.

File: dev/tests/integration/testsuite/Magento/ProductAlert/_files/customer_unsubscribe_stock_rollback.php [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+ */
+
+use Magento\Catalog\Api\ProductRepositoryInterface;
+use Magento\ProductAlert\Model\ResourceModel\Stock;
+use Magento\TestFramework\Helper\Bootstrap;
+
+$objectManager = Bootstrap::getObjectManager();
+$resource = $objectManager->get(Stock::class);
+
+$productRepository = $objectManager->create(ProductRepositoryInterface::class);
```
Comment (phoenix128) [inline dev/tests/integration/testsuite/Magento/ProductAlert/_files/customer_unsubscribe_stock_rollback.php]: Why are you using `create`?

**Review by sidolov — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @sidolov, thank you for the review.
[ENGCOM-5940](https://jira.corp.magento.com/browse/ENGCOM-5940) has been created to process this Pull Request
:eight_spoked_asterisk: @sidolov, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

File: dev/tests/integration/testsuite/Magento/ProductAlert/_files/customer_unsubscribe_stock_rollback.php (hunk 2 above)
Comment (yuriichayka) [reply dev/tests/integration/testsuite/Magento/ProductAlert/_files/customer_unsubscribe_stock_rollback.php]: Hello, thanks for pointing that out. Was changed.

File: dev/tests/integration/testsuite/Magento/ProductAlert/_files/customer_unsubscribe_stock_rollback.php (hunk 3 above)
Comment (yuriichayka) [reply dev/tests/integration/testsuite/Magento/ProductAlert/_files/customer_unsubscribe_stock_rollback.php]: Changed to 'get'. Should have been that way initially, result of copypasting

**Pull request merged**